### DISCUSSION TITLE
Using xid or ksuid

### DISCUSSION THREAD

**Discussion by barbinbrad** [supabase/supabase#11812] [Questions] [answered]
Read an interesting blog post this morning on database keys. Wanted to try out `xid` instead of `uuid` (per the recommendation of the article). But I'm unsure of how to install it.

For uuid I run `CREATE EXTENSION IF NOT EXISTS "uuid-ossp";` 

Is there a similar command I can run to get access to `xid` or `ksuid`?

**Comment by barbinbrad** [comment]
Or do I just run this SQL script? 

https://github.com/modfin/pg-xid/blob/master/xid.sql

**Comment by GaryAustin1** [comment, marked as answer]
If it does not have any dependencies and is just that sql code then you should just be able to run that to "install" it.

Is this the blog post you read?  https://supabase.com/blog/choosing-a-postgres-primary-key

**Comment by peugene** [comment]
Thanks for this Q/A,
I run the script xid.sql by just replacing public. by extensions.
Now i can use xid as primary key : id extensions.xid default extensions.xid() primary key
